src/sage/modular/modform/element.py

```diff
@@ -233,9 +265,111 @@ def __call__(self, x, prec=None):
 
             sage: f(0)
             0
-        """
+
+        Evaluate numerically::
+
+            sage: f = ModularForms(1, 12).0
+            sage: f(0.3)  # rel tol 1e-12
+            2.3452457654859093943911095448677943331e-6
+            sage: f = EisensteinForms(1, 4).0
+            sage: f(0.9)  # rel tol 1e-12
+            1.2647594220924141255379137476604196202e7
+
+        TESTS::
+
+            sage: f = ModularForms(96, 2).0
+            sage: f(0.3)  # rel tol 1e-12
+            0.299999997396191
+            sage: f(0.0+0.0*I)
+            0
+
+        Higher precision::
+
+            sage: f(ComplexField(1024)(0.3))  # rel tol 1e-300
+            0.299999997396191310292851660587501640585287966606926712372685051052183848485101271417891225951099656119402111562859500035032541015715233392211176407057239620967660312966086779337994327624536037187214146041831261633868799976346085552246983798477217725916938994733822571999849908529513379575860920140944745353698614348900664409173
+            sage: f(ComplexField(1024)(1+2*I)/3)  # rel tol 1e-300
+            0.321653845723568825567905326693899006909160675826045434571915911867833071589696569615194893689901721899490426373845744862497953969333193682758406183041097473225418603897622369265087989491842075279034290073993079800285909375526425931622610153626461152157393613063055863371355916362145445869467929251660349223775541765403069287756 + 0.670612446383675863028207907112577773897857202143717552376205630684889727243085419317012486309756101468259545944957593645365938699136657674117927754488042563377649540662343013944001734211314704342435690945950614217421684269575557390763317106676805075226392963322227027538349081598725382171963223815408532667527371217267163311545*I
```

Comment:
Could you split this at the `+`? Similarly below. (Although do we really need a test at such high precision?)

Reply:
Maybe not 1024, but I think at least it should be 128 to ensure there isn't any accidental conversion to `CC` (53 bits) along the way.